traffic_ops/traffic_ops_golang/api_capability/api_capabilities.go

```diff
@@ -0,0 +1,110 @@
+package api_capability
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/apache/trafficcontrol/lib/go-tc"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/dbhelpers"
+	"github.com/jmoiron/sqlx"
+)
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+func GetAPICapabilitiesHandler(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	results, usrErr, sysErr := getAPICapabilities(inf.Tx, inf.Params)
+
+	if usrErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, usrErr, nil)
+		return
+	}
+
+	if sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, nil, sysErr)
+		return
+	}
```

Comment:
A system error is implicitly server-side, so you'll want to use a more appropriate response code in that case, like `http.StatusInternalServerError`.

But also sometimes both a user error and system error can be helpful, and in any case this can be simplified if you reduce it to:

```go
results, userErr, sysErr, errCode := getAPICapabilities(inf.Tx, inf.Params)
if userErr != nil || sysErr != nil {
	api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
	return
}
```
And then obviously change `getAPICapabilities` to return an appropriate response code. Otherwise in order to support the possibility of both being non-`nil` and return the proper response code you'll need to do some more involved checking. I know your function doesn't do that now, but the change would make it more robust and people could add functionality more safely.